bindings/node/index.d.ts

```diff
@@ -191,6 +191,11 @@ export interface ClientEncryptionOptions {
    */
   kmsProviders?: KMSProviders;
 
+  /**
+   * Optional callback to override KMS providers per-context.
+   */
+  onKmsProviderRefresh?: () => KMSProviders | Promise<KMSProviders>;
```

Comment:
This can be both sync or async? Could we just make it always async?

Reply:
If we ensure it always returns a promise then it's safe to call `.then` on it, which I know isn't being done here but it's a nice guarantee I think

Reply:
Sure, done :)